### PR TITLE
chore(CPlusPlus): add Digit Dynamic Programming

### PR DESCRIPTION
**Pull request by Masterroy1210**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**

Added  two Standard Digit Dynamic Programming problem in the DIGIT DP section

**Other information:**

### CODE REVIEW

**Comment by ming-tsai** [comment]
Hi @Masterroy1210,
What is different between Dynamic Programming and Digit Dynamic Programing?

**Comment by Masterroy1210** [comment]
> Hi @Masterroy1210,
> What is different between Dynamic Programming and Digit Dynamic Programing?

Traditional dp is like memonizing the ans for overlapping substructure so that we do not calculate it again 
But  in digit dp we jst make required number by putting a digit one by one and keep track using flags

File: algorithms/CPlusPlus/Digit Dynamic Programming/Digit-dp-2.cpp [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        ans.push_back(n%10);
+        n=n/10;
+    }
+    reverse(ans.begin(),ans.end());
+    return ans;
+
+}
+
+//Recusrsive digit dp function that forms the number between range o to nums;
+
```
Comment (aayushjain7) [inline algorithms/CPlusPlus/Digit Dynamic Programming/Digit-dp-2.cpp:25]: Correct the spelling for Recusrsive.

File: algorithms/CPlusPlus/Digit Dynamic Programming/Digit-dp-2.cpp [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        //if first is not equal to last we jst return 0;
+        else{
+            return 0;
+        }
+    }
+
+    //Initially limit is 9 that means we call fill digit from 0 to 9;
+    int limit =9;
+    ////if flag==0 means that digit is smaller already and it should not be made greated and hence 
+    //limit should be nums[index];
```
Comment (aayushjain7) [inline algorithms/CPlusPlus/Digit Dynamic Programming/Digit-dp-2.cpp:43]: change greated to greater.

**Review by aayushjain7 — CHANGES_REQUESTED** [review]

**Comment by ming-tsai** [comment]
Hi @Masterroy1210,
Any updated?

**Comment by ming-tsai** [comment]
Hi @Masterroy1210,
This pull request will be closed, any updates please re-open this pull request.

**Pull request closed without merging**